app/code/core/Mage/SalesRule/Model/Observer.php

```diff
@@ -120,7 +120,6 @@ public function sales_order_afterPlace($observer)
                 }
             }
             $coupon = Mage::getModel('salesrule/coupon');
-            /** @var Mage_SalesRule_Model_Coupon */
```

Comment:
Can this be moved one line above.

Reply:
This comment is not needed. Both phpstan and phpstorm know type of this variable.

Reply:
This comment is not needed. Both phpstan and phpstorm knows type of this variable.